core/templates/hashfuncs.h

```diff
@@ -201,17 +201,18 @@ static _FORCE_INLINE_ uint32_t hash_fmix32(uint32_t h) {
 static _FORCE_INLINE_ uint32_t hash_murmur3_buffer(const void *key, int length, const uint32_t seed = HASH_MURMUR3_SEED) {
 	// Although not required, this is a random prime number.
 	const uint8_t *data = (const uint8_t *)key;
-	const int nblocks = length / 4;
+	const int nblocks = length / sizeof(uint32_t);
 
 	uint32_t h1 = seed;
 
 	const uint32_t c1 = 0xcc9e2d51;
 	const uint32_t c2 = 0x1b873593;
 
-	const uint32_t *blocks = (const uint32_t *)(data + nblocks * 4);
+	const uint8_t *blocks = data + nblocks * sizeof(uint32_t);
 
 	for (int i = -nblocks; i; i++) {
-		uint32_t k1 = blocks[i];
+		uint32_t k1;
+		memcpy(&k1, &blocks[i * sizeof(uint32_t)], sizeof(uint32_t));
```

Comment:
I think it would be good to add a comment here to explain why we don't just cast to const `uint32_t *`

Reply:
Done. I originally did this just to match the language spec "[a char, unsigned char, or std​::​byte type](https://eel.is/c++draft/basic.lval#11.sentence-1)", but it's not realistically going to be a problem.